Docs 4067 ha cluster

Cristian-Venticinque opened the pull request:
Added message stating that clusters are not available in CloudHub. 
Also restructured some content, and applied styling to improve readability.

Commit:
Added limitation: Clustering is not available in CloudHub.

Commit:
Corrected xref to CloudHub, it redirected to Runtime Manager.

Commit:
Added CloudHub HA and DR to See Also links.

Commit:
Added [IMPORTANT] at the top: Clusters are not available in CH

Commit:
Removed word "About" from titles. Fixed styling.

Commit:
Removed "Contact Support" from article.

Commit:
Removed "SEDA Queues". These don't exist in Mule 4.

Commit:
Moved Prerequisites to top. Removed limitations.

Requisites and limitations were in the same paragraph, I separated them 
and moved prerequisites to the top since this should be introduced 
first. 
Limitations were actually a copy of some senteces already mentioned in 
the article.

Commit:
Removed conflicting statement. 

-Since we recommend to avoid having more than 8 Mule nodes, we cannot 
also recommend to create as many nodes as needed.

Commit:
Clarified a sentence. Added proper styling to IP and ports.

dukesphere commented on modules/ROOT/pages/mule-high-availability-ha-clusters.adoc:
```diff
@@ -4,6 +4,9 @@ include::_attributes.adoc[]
+[IMPORTANT]
+Clustering is not available in xref:runtime-manager::cloudhub.adoc[CloudHub], see xref:runtime-manager::cloudhub-fabric.adoc[CloudHub Fabric] for details of how workers can be shared or doubled to scale your application and provide high availability.
+
 A *cluster* is a set of Mule runtimes that acts as a unit. In other words, a cluster is a virtual server composed of multiple nodes. The nodes (Mule runtimes) in a cluster communicate and share information through a distributed shared memory grid. This means that the data is replicated across memory in different physical machines.
```
Please go ahead and remove bold font from anything that's not a GUI label. Or Mysti will have to do it before we get it translated.

dukesphere replied:
Pls remove bold (e.g., `*cluster*`) for anything that's not a GUI label. Mysti will have to do it otherwise when she cleans up docs for translation.

dukesphere commented on modules/ROOT/pages/mule-high-availability-ha-clusters.adoc:
```diff
@@ -52,11 +55,16 @@ The following problems may exist when you have a server group composed of multip
 
+== Prerequisites
 
+* You must have at least two Mule runtimes in a cluster, each of which should run on different physical (or virtual) machines.
```
Pls change "should" to something else. Maybe "cluster, where each runs on" or something like that.  We're supposed to avoid "should".

dukesphere commented on modules/ROOT/pages/mule-high-availability-ha-clusters.adoc:
```diff
@@ -52,11 +55,16 @@ The following problems may exist when you have a server group composed of multip
+== Clustering
 
-With out of the box configurations, it is recommended that you scale a cluster to no more than eight Mule runtimes. All the Mule runtimes in a cluster group together to form a single unit. Thus, you can deploy, monitor, or stop all the Mule runtimes in a cluster as if they were a single Mule runtime. If you need more nodes than eight, contact https://www.mulesoft.com/support-and-services/mule-esb ...
+With out of the box configurations, it is recommended that you scale a cluster to no more than eight Mule runtimes. All the Mule runtimes in a cluster group together to form a single unit. Thus, you can deploy, monitor, or stop all the Mule runtimes in a cluster as if they were a single Mule runtime.
```
I think we're supposed to avoid "out of the box" and "out-of-the-box" (even though it's everywhere in our docs).  Maybe "default configurations"? I'm not sure what the best replacement is here.

dukesphere reviewed (commented):
suggesting changes to conform with our style guide conventions

Commit:
Removed incorrect *bold* styling.

Fixed some titles by adding spaces and proper structure to some 
sentences.

Commit:
Replaced usage of "should" and "out of the box".

fermujica commented on modules/ROOT/pages/mule-high-availability-ha-clusters.adoc:
```diff
@@ -4,7 +4,10 @@ include::_attributes.adoc[]
 
-A *cluster* is a set of Mule runtimes that acts as a unit. In other words, a cluster is a virtual server composed of multiple nodes. The nodes (Mule runtimes) in a cluster communicate and share information through a distributed shared memory grid. This means that the data is replicated across memory in different physical machines.
+[IMPORTANT]
+Clustering is not available in xref:runtime-manager::cloudhub.adoc[CloudHub], see xref:runtime-manager::cloudhub-fabric.adoc[CloudHub Fabric] for details of how workers can be shared or doubled to scale your application and provide high availability.
```
I don't think these inline links to CloudHub ~~and CloudHub Fabric~~ are necessary here. 
**EDIT**: On a second thought, CloudHub Fabric link actually makes sense in this line.
At this point we assume the user is familiar with those products. 
In this particular case, I think these links would make more sense in the `See Also` section.

Commit:
Removed external links. Added `backticks` to class name.

Commit:
Moved inline xrefs to See Also section.

Commit:
Removed unnecessary internal links.

Commit:
Removed unnecesarry inline xrefs, restructured some content.

dukesphere commented on modules/ROOT/pages/mule-high-availability-ha-clusters.adoc:
```diff
@@ -4,7 +4,10 @@ include::_attributes.adoc[]
 
-A *cluster* is a set of Mule runtimes that acts as a unit. In other words, a cluster is a virtual server composed of multiple nodes. The nodes (Mule runtimes) in a cluster communicate and share information through a distributed shared memory grid. This means that the data is replicated across memory in different physical machines.
+[IMPORTANT]
+Clustering is not available in CloudHub, see xref:runtime-manager::cloudhub-fabric.adoc[CloudHub Fabric] for details of how workers can be shared or doubled to scale your application and provide high availability.
```
Cloudhub, see ... details of -> CloudHub. See ... details on

dukesphere commented on modules/ROOT/pages/mule-high-availability-ha-clusters.adoc:
```diff
@@ -4,7 +4,10 @@ include::_attributes.adoc[]
+[IMPORTANT]
+Clustering is not available in CloudHub, see xref:runtime-manager::cloudhub-fabric.adoc[CloudHub Fabric] for details of how workers can be shared or doubled to scale your application and provide high availability.
+
+A Cluster is a set of Mule runtimes that acts as a unit. In other words, a cluster is a virtual server composed of multiple nodes. The nodes (Mule runtimes) in a cluster communicate and share information through a distributed shared memory grid. This means that the data is replicated across memory in different physical machines.
```
A Cluster -> A cluster 

Mule runtimes -> Mule runtime engines      // note from sDuke: we have to use "Mule" or "Mule runtime engine" everywhere

dukesphere commented on modules/ROOT/pages/mule-high-availability-ha-clusters.adoc:
```diff
@@ -13,7 +16,7 @@ Contact your customer service representative about pricing for this feature.
 == The Benefits of Clustering
 
-By default, clustering Mule runtimes ensures high system availability. (If you wish to use clusters for high performance instead, see <<Clustering for High Performance>> below.) If a Mule runtime node becomes unavailable due to failure or planned downtime, another node in the cluster can assume the workload and continue to process existing events and messages. The following figure illustrates the ...
+By default, clustering Mule runtimes ensures high system availability. If a Mule runtime node becomes unavailable due to failure or planned downtime, another node in the cluster can assume the workload and continue to process existing events and messages. The following figure illustrates the processing of incoming messages by a cluster of two nodes. Notice that the processing load is balanced acr ...
```
same everywhere: Mule runtimes -> Mule runtime engines

You might use a colon (:) before "Node 1" instead of a `-`, which is less commonly understood (IMHO)

dukesphere commented on modules/ROOT/pages/mule-high-availability-ha-clusters.adoc:
```diff
@@ -29,79 +32,96 @@ image::cluster-diagram.png[]
-* Automatic coordination of access to resources such as files, databases, and FTP sources. The Mule runtime cluster automatically manages which node (Mule runtime) will handle communication from a data source.
-* Automatic load balancing of processing within a cluster. If you divide your flows into a series of steps and connect these steps with a transport such as VM, each step is put in a queue, making it cluster enabled. The cluster of Mule runtimes can then process each step in any node, and so better balance the load across nodes.
-* Raised alerts. You can set up an alert to appear when a node goes down and when a node comes back up.
+* Automatic coordination of access to resources such as files, databases, and FTP sources. +
```
resources such as -> resources, such as 
(I had to look it up: https://www.grammarly.com/blog/such-as-comma/)

dukesphere commented on modules/ROOT/pages/mule-high-availability-ha-clusters.adoc:
```diff
@@ -29,79 +32,96 @@ image::cluster-diagram.png[]
-* Automatic load balancing of processing within a cluster. If you divide your flows into a series of steps and connect these steps with a transport such as VM, each step is put in a queue, making it cluster enabled. The cluster of Mule runtimes can then process each step in any node, and so better balance the load across nodes.
-* Raised alerts. You can set up an alert to appear when a node goes down and when a node comes back up.
+* Automatic coordination of access to resources such as files, databases, and FTP sources. +
+The Mule runtime cluster automatically manages which node (Mule runtime) will handle communication from a data source.
```
Mule runtime engine

dukesphere commented on modules/ROOT/pages/mule-high-availability-ha-clusters.adoc:
```diff
@@ -29,79 +32,96 @@ image::cluster-diagram.png[]
+* Automatic coordination of access to resources such as files, databases, and FTP sources. +
+The Mule runtime cluster automatically manages which node (Mule runtime) will handle communication from a data source.
+* Automatic load balancing of processing within a cluster. +
+If you divide your flows into a series of steps and connect these steps with a transport such as VM, each step is put in a queue, making it cluster enabled. The cluster of Mule runtimes can then process each step in any node, and so better balance the load across nodes.
```
cluster enabled -> cluster-enabled

dukesphere replied:
Mule runtime engine

Suggested (comma not needed if you keep the original):
, and so better -> to better balance

dukesphere commented on modules/ROOT/pages/mule-high-availability-ha-clusters.adoc:
```diff
@@ -29,79 +32,96 @@ image::cluster-diagram.png[]
 === Concurrency Issues Solved by Clusters
 
-The following problems may exist when you have a server group composed of multiple servers that *aren't binded as a cluster*. You don't have to worry about any of them if you group your servers as a cluster:
+The following problems may exist when you have a server group composed of multiple servers that aren't bound as a cluster. You don't have to worry about any of them if you group your servers as a cluster:
```
may -> might 
(it's one of those tech writing standards)

dukesphere commented on modules/ROOT/pages/mule-high-availability-ha-clusters.adoc:
```diff
@@ -29,79 +32,96 @@ image::cluster-diagram.png[]
 
-* *File based transports*: All Mule instances access the same mule file folders concurrently, which can lead to duplicate file processing and even possible failures if a file is deleted or modified by the Mule application
+* File based transports. +
+All Mule instances access the same mule file folders concurrently, which can lead to duplicate file processing and even possible failures if a file is deleted or modified by the Mule application.
```
mule -> Mule

and even possible -> and possible

dukesphere commented on modules/ROOT/pages/mule-high-availability-ha-clusters.adoc:
```diff
@@ -29,79 +32,96 @@ image::cluster-diagram.png[]
 
-* *Multicast transport*: All mule instances get the same TCP requests and then process duplicate messages
+* Multicast transport. +
+All mule instances get the same TCP requests and then process duplicate messages.
```
mule -> Mule

dukesphere commented on modules/ROOT/pages/mule-high-availability-ha-clusters.adoc:
```diff
@@ -29,79 +32,96 @@ image::cluster-diagram.png[]
+All mule instances get the same TCP requests and then process duplicate messages.
 
-* *JMS Topics*: All mule instances connect to the same JMS topic, which may lead to repeated processing of messages when scaling the non clustered Mule instance horizontally
+* JMS Topics. +
```
Topics -> topics

dukesphere commented on modules/ROOT/pages/mule-high-availability-ha-clusters.adoc:
```diff
@@ -29,79 +32,96 @@ image::cluster-diagram.png[]
 
-* *JMS Topics*: All mule instances connect to the same JMS topic, which may lead to repeated processing of messages when scaling the non clustered Mule instance horizontally
+* JMS Topics. +
+All mule instances connect to the same JMS topic, which may lead to repeated processing of messages when scaling the non clustered Mule instance horizontally.
```
mule -> Mule //from sDuke: should be capitalized everywhere

dukesphere replied:
may -> might //likely everywhere it appears

dukesphere replied:
non clustered -> non-clustered //or nonclustered if you follow the Chicago Manual of Style...

dukesphere commented on modules/ROOT/pages/mule-high-availability-ha-clusters.adoc:
```diff
@@ -29,79 +32,96 @@ image::cluster-diagram.png[]
 
-* *JMS request-reply/request-response*: All Mule instances are listening for messages in the same response queue, this implies that a Mule instance might obtain a response that isn't correlated to the request it sent. This can result in incorrect responses or make a flow fail with timeout.
+* JMS request-reply/request-response. +
+All Mule instances are listening for messages in the same response queue, this implies that a Mule instance might obtain a response that isn't correlated to the request it sent. This can result in incorrect responses or make a flow fail with timeout.
```
queue, this -> queue, which 

it sent. This can result in incorrect...  -> it sent and might result in incorrect...

// In general, we want "this" to modify a noun. I recommend removing them in this case and rewording something like suggested above.

dukesphere commented on modules/ROOT/pages/mule-high-availability-ha-clusters.adoc:
```diff
@@ -29,79 +32,96 @@ image::cluster-diagram.png[]
+All Mule instances are listening for messages in the same response queue, this implies that a Mule instance might obtain a response that isn't correlated to the request it sent. This can result in incorrect responses or make a flow fail with timeout.
 
-* *Idempotent-redelivery-policy*: Idempotency doesn’t work if the same request is received by different Mule instances. Duplicated messages aren’t possible
+* Idempotent-redelivery-policy. +
```
Why is "Idempotent-redelivery-policy" hyphenated. If its an XML element or something, it shouldn't be capitalized. I'd recommend changing it to Indempotent Redelivery Policy. If it's an XML element, you could add a parenthetical (`indempotent-redelivery-policy`) after that.

dukesphere commented on modules/ROOT/pages/mule-high-availability-ha-clusters.adoc:
```diff
@@ -29,79 +32,96 @@ image::cluster-diagram.png[]
 
-* *Salesforce streaming API*: if multiple instances of the same application are deployed, they will fail since the API only supports a single consumer. No failover support in case the instance connected is stopped or crashes
+* Salesforce streaming API. +
+if multiple instances of the same application are deployed, they will fail since the API only supports a single consumer. No failover support in case the instance connected is stopped or crashes.
```
if -> If

"since" -> "because" 
// It's a tech writing thing because "since" can indicate time or causality, whereas "because" only indicates causality. Not always followed by writers, but it's standard in the industry. 

No failover support in case the instance connected... -> There is no failover support if the connected instance...

dukesphere commented on modules/ROOT/pages/mule-high-availability-ha-clusters.adoc:
```diff
@@ -29,79 +32,96 @@ image::cluster-diagram.png[]
 
+== Prerequisites
 
+* You must have at least two Mule runtimes in a cluster, each one running on different physical (or virtual) machines.
```
Mule runtime engines

dukesphere commented on modules/ROOT/pages/mule-high-availability-ha-clusters.adoc:
```diff
@@ -29,79 +32,96 @@ image::cluster-diagram.png[]
+== Prerequisites
 
+* You must have at least two Mule runtimes in a cluster, each one running on different physical (or virtual) machines.
+* Mule HA requires a reliable network connection between servers to maintain synchronization between the nodes in the cluster.
```
HA -> high availability (HA)

dukesphere commented on modules/ROOT/pages/mule-high-availability-ha-clusters.adoc:
```diff
@@ -29,79 +32,96 @@ image::cluster-diagram.png[]
+== Clustering
 
-With out of the box configurations, it is recommended that you scale a cluster to no more than eight Mule runtimes. All the Mule runtimes in a cluster group together to form a single unit. Thus, you can deploy, monitor, or stop all the Mule runtimes in a cluster as if they were a single Mule runtime. If you need more nodes than eight, contact https://www.mulesoft.com/support-and-services/mule-esb ...
+With the initial configuration, we recommend that you scale a cluster to no more than eight Mule runtimes. All the Mule runtimes in a cluster group together to form a single unit. Thus, you can deploy, monitor, or stop all the Mule runtimes in a cluster as if they were a single Mule runtime.
```
we recommend -> MuleSoft recommends
// typically avoid "we" except in tutorials

Mule runtime engines 

to form -> form

dukesphere commented on modules/ROOT/pages/mule-high-availability-ha-clusters.adoc:
```diff
@@ -29,79 +32,96 @@ image::cluster-diagram.png[]
-With out of the box configurations, it is recommended that you scale a cluster to no more than eight Mule runtimes. All the Mule runtimes in a cluster group together to form a single unit. Thus, you can deploy, monitor, or stop all the Mule runtimes in a cluster as if they were a single Mule runtime. If you need more nodes than eight, contact https://www.mulesoft.com/support-and-services/mule-esb ...
+With the initial configuration, we recommend that you scale a cluster to no more than eight Mule runtimes. All the Mule runtimes in a cluster group together to form a single unit. Thus, you can deploy, monitor, or stop all the Mule runtimes in a cluster as if they were a single Mule runtime.
 
 All the Mule runtimes in a cluster share memory, as illustrated below:
```
Mule runtime engines

dukesphere commented on modules/ROOT/pages/mule-high-availability-ha-clusters.adoc:
```diff
@@ -29,79 +32,96 @@ image::cluster-diagram.png[]
 image::topology-4-cluster.png[topology_4-cluster]
 
-Mule uses an *active-active* model to cluster Mule runtimes, rather than an *active-passive* model.
+Mule uses an active-active model to cluster Mule runtimes, rather than an active-passive model.
```
Mule runtime engines

dukesphere commented on modules/ROOT/pages/mule-high-availability-ha-clusters.adoc:
```diff
@@ -29,79 +32,96 @@ image::cluster-diagram.png[]
+Mule uses an active-active model to cluster Mule runtimes, rather than an active-passive model.
 
-In an *active-passive* model, one node in a cluster acts as the *primary*, or active node, while the others are *secondary*, or passive nodes. The application in such a model runs on the primary node, and only ever runs on the secondary node if the first one fails. In this model, the processing power of the secondary node(s) is mostly wasted in passive waiting for the primary node to fail.
+In an active-passive model, one node in a cluster acts as the primary, or active node, while the others are secondary, or passive nodes. The application in such a model runs on the primary node, and only ever runs on the secondary node if the first one fails. In this model, the processing power of the secondary node(s) is mostly wasted in passive waiting for the primary node to fail.
```
node, and only ever -> node and only 
// no comma is needed because these are not independent clauses

dukesphere replied:
wasted in passive waiting -> wasted in passively waiting

dukesphere commented on modules/ROOT/pages/mule-high-availability-ha-clusters.adoc:
```diff
@@ -29,79 +32,96 @@ image::cluster-diagram.png[]
+In an active-passive model, one node in a cluster acts as the primary, or active node, while the others are secondary, or passive nodes. The application in such a model runs on the primary node, and only ever runs on the secondary node if the first one fails. In this model, the processing power of the secondary node(s) is mostly wasted in passive waiting for the primary node to fail.
 
-In an *active-active* model, no one node in the cluster acts as the primary node; all nodes in the cluster support the application. This application in this model runs on all the nodes, even splitting apart message processing between nodes to expedite processing across nodes.
+In an active-active model, no one node in the cluster acts as the primary node; all nodes in the cluster support the application. This application in this model runs on all the nodes, even splitting apart message processing between nodes to expedite processing across nodes.
```
This application -> The application

nodes, even splitting apart message -> nodes and splits message

dukesphere commented on modules/ROOT/pages/mule-high-availability-ha-clusters.adoc:
```diff
@@ -29,79 +32,96 @@ image::cluster-diagram.png[]
+== Queues
 
-You can set up a *VM queue* explicitly to *load balance* across Mule runtimes (nodes). Thus, if your entire application flow contains a sequence of child flows, the cluster can assign each successive child flow to whichever Mule runtime (node) happens to be available at the time. Potentially, the cluster can process a single message on multiple nodes as it passes through the VM endpoints in the a ...
+You can set up a VM queue explicitly to load balance across Mule runtimes (nodes). Thus, if your entire application flow contains a sequence of child flows, the cluster can assign each successive child flow to whichever Mule runtime (node) happens to be available at the time. Potentially, the cluster can process a single message on multiple nodes as it passes through the VM endpoints in the appli ...
```
Mule runtime engines

dukesphere commented on modules/ROOT/pages/mule-high-availability-ha-clusters.adoc:
```diff
@@ -29,79 +32,96 @@ image::cluster-diagram.png[]
 . Highly reliable individual connections
 
-The feature known as *transactionality* tracks application event sequences to ensure that each message-processing step gets completed successfully, and therefore, no messages get lost or processed incorrectly. If a step fails, for some reason, the transactional mechanism rolls back all previous processing events, then restarts the message processing sequence again.
+The feature known as transactionality tracks application event sequences to ensure that each message-processing step gets completed successfully, and therefore, no messages get lost or processed incorrectly. If a step fails, for some reason, the transactional mechanism rolls back all previous processing events, then restarts the message processing sequence again.
```
gets completed successfully -> completes successfully

fails, for some reason -> fails for some reason

dukesphere commented on modules/ROOT/pages/mule-high-availability-ha-clusters.adoc:
```diff
@@ -29,79 +32,96 @@ image::cluster-diagram.png[]
 Transports such as JMS, VM, and JDBC provide built-in transactional support, thus ensuring that messages get processed reliably. For example, you can configure a transaction on an inbound JMS connection endpoint to remove messages from the JMS server only after the transaction has been committed. This ensures that the original message remains available for reprocessing if an error occurs during t ...
 
-You must use *XA transactions* to move messages between dissimilar transports that support transactions. This ensures that the Mule runtime commits associated transactions from all the dissimilar transports as a single unit. See xref:transaction-management.adoc[Transaction Management] for more information on XA and other types of transactions supported by Mule runtimes.
+You must use XA transactions to move messages between dissimilar transports that support transactions. This ensures that the Mule runtime commits associated transactions from all the dissimilar transports as a single unit.
```
//Please follow up on this tech question: Is this talking about Mule transports? If so, we don't have Mule transports in Mule 4, though transport protocols are supported. I'm not sure if XA transactions are supported in Mule 4 either. For some background info, see https://docs.mulesoft.com/mule-runtime/4.1/migration-transports

dukesphere replied:
This ensures -> This practice ensures
//note: In general, "This" should modify something. 

Mule runtime engine

Cristian-Venticinque replied:
@aiannucci , could you please confirm if this section is talking about transport protocols or specific Mule Transports from 3.x?

dukesphere commented on modules/ROOT/pages/mule-high-availability-ha-clusters.adoc:
```diff
@@ -29,79 +32,96 @@ image::cluster-diagram.png[]
-You must use *XA transactions* to move messages between dissimilar transports that support transactions. This ensures that the Mule runtime commits associated transactions from all the dissimilar transports as a single unit. See xref:transaction-management.adoc[Transaction Management] for more information on XA and other types of transactions supported by Mule runtimes.
+You must use XA transactions to move messages between dissimilar transports that support transactions. This ensures that the Mule runtime commits associated transactions from all the dissimilar transports as a single unit.
 
 == Cluster Support for Transports
```
//tech issue re "Cluster Support for Transports": Pls follow up on "Transports". Not sure if this section is up-to-date for Mule 4.

aiannucci replied:
Yeah, transport concept is old, we must rewrite this

dukesphere commented on modules/ROOT/pages/mule-high-availability-ha-clusters.adoc:
```diff
@@ -29,79 +32,96 @@ image::cluster-diagram.png[]
-* Socket-based transports read input sent to network sockets that Mule owns. Examples include TCP, UDP, and HTTP[S].
-* listener-based transports read data using a protocol that fully supports concurrent multiple accessors. Examples include JMS and VM.
-* resource-based transports read data from a resource that allows multiple concurrent accessors, but does not natively coordinate their use of the resource. For instance, suppose multiple programs are processing files in the same shared directory by reading, processing, and then deleting the files. These programs must use an explicit, application-level locking strategy to prevent the same file fr ...
+* Socket-based transports: reads input sent to network sockets that Mule owns. +
```
reads -> read

Mule owns //"owns": I'm not sure about that terminology. Not sure about replacement term.

dukesphere commented on modules/ROOT/pages/mule-high-availability-ha-clusters.adoc:
```diff
@@ -29,79 +32,96 @@ image::cluster-diagram.png[]
-* resource-based transports read data from a resource that allows multiple concurrent accessors, but does not natively coordinate their use of the resource. For instance, suppose multiple programs are processing files in the same shared directory by reading, processing, and then deleting the files. These programs must use an explicit, application-level locking strategy to prevent the same file fr ...
+* Socket-based transports: reads input sent to network sockets that Mule owns. +
+Examples include TCP, UDP, and HTTP[S].
+* Listener-based transports: reads data using a protocol that fully supports concurrent multiple accessors. +
```
reads -> read

dukesphere commented on modules/ROOT/pages/mule-high-availability-ha-clusters.adoc:
```diff
@@ -29,79 +32,96 @@ image::cluster-diagram.png[]
+Examples include TCP, UDP, and HTTP[S].
+* Listener-based transports: reads data using a protocol that fully supports concurrent multiple accessors. +
+Examples include JMS and VM.
+* Resource-based transports: reads data from a resource that allows multiple concurrent accessors, but does not natively coordinate their use of the resource. +
```
reads -> read

dukesphere commented on modules/ROOT/pages/mule-high-availability-ha-clusters.adoc:
```diff
@@ -29,79 +32,96 @@ image::cluster-diagram.png[]
+* Listener-based transports: reads data using a protocol that fully supports concurrent multiple accessors. +
+Examples include JMS and VM.
+* Resource-based transports: reads data from a resource that allows multiple concurrent accessors, but does not natively coordinate their use of the resource. +
+For instance, suppose multiple programs are processing files in the same shared directory by reading, processing, and then deleting the files. These programs must use an explicit, application-level locking strategy to prevent the same file from being processed more than once. Examples of resource-based transports include File, FTP, SFTP, E-mail, and JDBC.
```
// note: Mule 3 transports were replaced by connectors in Mule 4. I wonder if we need to reword this to account for that or if we're talking about transport protocols. The JDBC transport would be subsumed within the Database connectors.

dukesphere replied:
Compare 3.9 doc at https://docs.mulesoft.com/mule-runtime/3.9/transports-reference with 4.1 connector docs at https://docs.mulesoft.com/connectors/

aiannucci replied:
Definitely need to talk about connectors instead of transports.

dukesphere commented on modules/ROOT/pages/mule-high-availability-ha-clusters.adoc:
```diff
@@ -29,79 +32,96 @@ image::cluster-diagram.png[]
 
 * Socket-based
-** Since each clustered Mule runtime runs on a different network node, each instance receives only the socket-based traffic sent to its node. Incoming socket-based traffic should be <<Clustering and Load Balancing>> to distribute it among the clustered instances.
+** Since each clustered Mule runtime runs on a different network node, each instance receives only the socket-based traffic sent to its node. Incoming socket-based traffic needs to apply Clustering and Load Balancing to distribute it among the clustered instances.
```
Since each -> Because each

Mule runtime engine

Clustering and Load Balancing -> clustering and load balancing

to distribute it -> //to distribute what? Please say what "it" is

dukesphere commented on modules/ROOT/pages/mule-high-availability-ha-clusters.adoc:
```diff
@@ -131,7 +151,7 @@ These actions can also apply to non-clustered applications.
 === Single Data-center Clustering
 
-To ensure reliable connectivity between cluster nodes, all nodes of a cluster should be located on the same LAN. Implementing a cluster with nodes across geographically separated locations, such as different data centers that are connected through a VPN, is possible but not recommended.
+To ensure reliable connectivity between cluster nodes, all nodes of a cluster have to be located on the same LAN. Implementing a cluster with nodes across geographically separated locations, such as different data centers that are connected through a VPN, is possible but not recommended.
```
have to be -> must be

dukesphere commented on modules/ROOT/pages/mule-high-availability-ha-clusters.adoc:
```diff
@@ -154,39 +174,37 @@ IMPORTANT: A cluster partition that does not have enough nodes to function will
 When Mule clusters are used to serve TCP requests (where TCP includes SSL/TLS, UDP, Multicast, HTTP, and HTTPS), some load balancing is needed to distribute the requests among the clustered instances. There are various software load balancers available, two of them are:
 
-* Nginx, an open-source HTTP server and reverse proxy. You can use nginx's http://wiki.nginx.org/HttpUpstreamModule[HttpUpstreamModule] for HTTP(S) load balancing. You can find further information in the Linode Library entry http://library.linode.com/web-servers/nginx/configuration/front-end-proxy-and-software-load-balancing[Use Nginx for Proxy Services and Software Load Balancing].
+* Nginx, an open-source HTTP server and reverse proxy. You can use nginx's `HttpUpstreamModule` for HTTP(S) load balancing.
```
Nginx -> NGINX
nginx's -> NGINX's

dukesphere replied:
https://www.nginx.com/ (just meant that I used that to confirm the spelling)

dukesphere commented on modules/ROOT/pages/mule-high-availability-ha-clusters.adoc:
```diff
@@ -154,39 +174,37 @@ IMPORTANT: A cluster partition that does not have enough nodes to function will
 == Clustering for High Performance
 
-Note that high performance is implemented differently on xref:runtime-manager::index.adoc[CloudHub] and xref:runtime-manager::deploying-to-pcf.adoc[Pivotal Cloud Foundry], so this section applies only for xref:runtime-manager::deploying-to-your-own-servers.adoc[on-premises deployments].
+Note that high performance is implemented differently on CloudHub and Pivotal Cloud Foundry, so this section applies only for on-premises deployments. +
```
//recommend rewording so that the reader sees immediately that the section is for on-premise deployments. People who scan docs will know quickly whether it applies to them or not. It's also unusual to begin a section with a note. I'm sure that note was a late addition to this section at some point.

dukesphere commented on modules/ROOT/pages/mule-high-availability-ha-clusters.adoc:
```diff
@@ -154,39 +174,37 @@ IMPORTANT: A cluster partition that does not have enough nodes to function will
 
-If high performance is your primary goal (rather than reliability), you can configure a Mule cluster or an individual application for maximum performance using a *performance profile*. By implementing the performance profile for specific applications within a cluster, you can maximize the scalability of your deployments while deploying applications with different performance and reliability requi ...
+
+If high performance is your primary goal (rather than reliability), you can configure a Mule cluster or an individual application for maximum performance using a performance profile. By implementing the performance profile for specific applications within a cluster, you can maximize the scalability of your deployments while deploying applications with different performance and reliability require ...
```
you apply it to all applications -> //you apply what to all applications? recommend rewording to something like: "Performance profiles that you configure at the container level apply to all applications within the container."

//The rewording aims to facilitate translation.

dukesphere commented on modules/ROOT/pages/mule-high-availability-ha-clusters.adoc:
```diff
@@ -154,39 +174,37 @@ IMPORTANT: A cluster partition that does not have enough nodes to function will
 * It implements the object store without backups, to avoid replication.
 
-To configure the performance profile at the _container_ level, add to *`mule-cluster.properties`* or to the system properties from the command line or wrapper.conf:
+To configure the performance profile at the _container_ level, add to `mule-cluster.properties` or to the system properties from the command line or wrapper.conf:
```
add to... -> add the following to `mule-cluster.properties`, to the system properties (from the command line), or to `wrapper.conf`:

dukesphere commented on modules/ROOT/pages/mule-high-availability-ha-clusters.adoc:
```diff
@@ -154,39 +174,37 @@ IMPORTANT: A cluster partition that does not have enough nodes to function will
 To configure the performance profile at the _individual application_ level, add the profile inside a configuration wrapper, as shown below.
 
-*Performance Store Profile*
+===== Performance Store Profile
```
//note: It's unusual to use a heading level in a case like this. Perhaps you could use a title notation, instead: 
===== Performance Store Profile -> .Performance Store Profile

dukesphere commented on modules/ROOT/pages/mule-high-availability-ha-clusters.adoc:
```diff
@@ -201,7 +219,7 @@ To configure the performance profile at the _individual application_ level, add
 Remember that application-level configuration overrides container-level configuration. If you would like to configure the container for high performance but make one ore more individual applications within that container prioritize reliability, include the following code in those applications:
 
-*Reliable Store Profile*
+===== Reliable Store Profile
```
===== Reliable Store Profile -> .Reliable Store Profile

dukesphere commented on modules/ROOT/pages/mule-high-availability-ha-clusters.adoc:
```diff
@@ -201,7 +219,7 @@ To configure the performance profile at the _individual application_ level, add
 
 Remember that application-level configuration overrides container-level configuration. If you would like to configure the container for high performance but make one ore more individual applications within that container prioritize reliability, include the following code in those applications:
```
that application-level -> that an application-level

overrides container-level -> overrides a container-level

would like -> want

ore -> or

dukesphere commented on modules/ROOT/pages/mule-high-availability-ha-clusters.adoc:
```diff
@@ -217,7 +235,7 @@ Remember that application-level configuration overrides container-level configur
 In cases of high load with endpoints that do not support load balancing, applying the performance profile may degrade performance. If you are using a File-based transport with an asynchronous processing strategy, JMS topics, multicasting, or HTTP connectors without a load balancer, the high volume of messages entering a single node can cause bottlenecks, and thus it can be better for performance  ...
 
-You can also choose to define a minimum number of machines required in a cluster for it to remain in an operational state. This grants you a consistency improvement. See  xref:creating-and-managing-a-cluster-manually.adoc#quorum-management[Quorum Management].
+You can also choose to define a minimum number of machines required in a cluster for it to remain in an operational state. This grants you a consistency improvement.
```
for it -> for the cluster

This grants you... // Are we saying "This configuration improves consistency" ? Should also indicate what sort of consistency we mean here. I'm not clear.

dukesphere reviewed (changes requested):
mostly style suggestions but also likely an update needed on the topic of transports

Commit:
Replaced "Mule runtime" with "Mule runtime engine"

Commit:
Fixed spelling and grammar errors. Applied proper styling.

-Also rewrote some sentences to improve readability and remove redundant 
words.

aiannucci commented on modules/ROOT/pages/mule-high-availability-ha-clusters.adoc:
```diff
@@ -4,7 +4,10 @@ include::_attributes.adoc[]
+[IMPORTANT]
+Clustering is not available in CloudHub, see xref:runtime-manager::cloudhub-fabric.adoc[CloudHub Fabric] for details on how workers can be shared or doubled to scale your application and provide high availability.
+
+A Cluster is a set of Mule runtime engines that acts as a unit. In other words, a cluster is a virtual server composed of multiple nodes. The nodes (Mule runtime engines) in a cluster communicate and share information through a distributed shared memory grid. This means that the data is replicated across memory in different physical machines.
```
```suggestion
A Cluster is a set of Mule runtime engines that acts as a unit. In other words, a cluster is a virtual server composed of multiple nodes. The nodes (Mule runtime engines) in a cluster communicate and share information through a distributed shared memory grid. This means that the data is replicated across memory in different machines.
```

aiannucci commented on modules/ROOT/pages/mule-high-availability-ha-clusters.adoc:
```diff
@@ -13,101 +16,118 @@ Contact your customer service representative about pricing for this feature.
 
 [NOTE]
-All Mule runtimes in a cluster actively process messages. Note that each Mule runtime is also internally scalable – a single Mule runtime can scale easily by taking advantage of multiple cores. Even when a Mule runtime takes advantage of multiple cores, it still operates as a single node (Mule runtime) in a cluster.
+All Mule runtime engines in a cluster actively process messages. Note that each Mule runtime engine is also internally scalable – a single Mule runtime engine can scale easily by taking advantage of multiple cores. Even when a Mule runtime engine takes advantage of multiple cores, it still operates as a single node (Mule runtime engine) in a cluster.
```
Can we rephrase to avoid repeating "Mule runtime engine" five times?

Cristian-Venticinque replied:
Please check if f62511e is ok.

aiannucci commented on modules/ROOT/pages/mule-high-availability-ha-clusters.adoc:
```diff
@@ -13,101 +16,118 @@ Contact your customer service representative about pricing for this feature.
 
+== Prerequisites
 
+* You must have at least two Mule runtime engines in a cluster, each one running on different physical (or virtual) machines.
```
```suggestion
* You must have at least two Mule runtime engines in a cluster, each one running on different machines.
```

aiannucci commented on modules/ROOT/pages/mule-high-availability-ha-clusters.adoc:
```diff
@@ -13,101 +16,118 @@ Contact your customer service representative about pricing for this feature.
+* You must have at least two Mule runtime engines in a cluster, each one running on different physical (or virtual) machines.
+* Mule high availability (HA) requires a reliable network connection between servers to maintain synchronization between the nodes in the cluster.
+* You must keep the following ports open to set up a Mule cluster: port `5701` and port `54327`.
+* You need to enable the multicast IP address: `224.2.2.3`. Multicast is used to discover new cluster members.
```
This actually depends on how you configure your cluster. We support both Multicast and TCP/IP config. So if you use TCP/IP, this item is not needed.

Cristian-Venticinque replied:
Please check if 5c7cdab is OK.

aiannucci commented on modules/ROOT/pages/mule-high-availability-ha-clusters.adoc:
```diff
@@ -13,101 +16,118 @@ Contact your customer service representative about pricing for this feature.
+== Clustering
 
-With out of the box configurations, it is recommended that you scale a cluster to no more than eight Mule runtimes. All the Mule runtimes in a cluster group together to form a single unit. Thus, you can deploy, monitor, or stop all the Mule runtimes in a cluster as if they were a single Mule runtime. If you need more nodes than eight, contact https://www.mulesoft.com/support-and-services/mule-esb ...
+With the initial configuration, MuleSoft recommends that you scale a cluster to no more than eight Mule runtime engines. All the Mule runtime engines in a cluster group together to form a single unit. Thus, you can deploy, monitor, or stop all the Mule runtime engines in a cluster as if they were a single Mule runtime engine.
```
Are these features of ARM? (deploy, monitor, stop all runtimes in a cluster)

Cristian-Venticinque replied:
This is also in the 3.9 article, so this was just moved along to 4.1 but can't confirm if it is referring to ARM.

aiannucci replied:
Ok, just want to confirm because these features are not natively supported by Mule, but maybe ARM provides them.

Cristian-Venticinque replied:
Who is able to confirm this?

Cristian-Venticinque replied:
Addressed in commit f73725d

aiannucci commented on modules/ROOT/pages/mule-high-availability-ha-clusters.adoc:
```diff
@@ -13,101 +16,118 @@ Contact your customer service representative about pricing for this feature.
+Mule uses an active-active model to cluster Mule runtime engines, rather than an active-passive model.
 
-In an *active-passive* model, one node in a cluster acts as the *primary*, or active node, while the others are *secondary*, or passive nodes. The application in such a model runs on the primary node, and only ever runs on the secondary node if the first one fails. In this model, the processing power of the secondary node(s) is mostly wasted in passive waiting for the primary node to fail.
+In an active-passive model, one node in a cluster acts as the primary, or active node, while the others are secondary, or passive nodes. The application in such a model runs on the primary node and only runs on the secondary node if the first one fails. In this model, the processing power of the secondary node(s) is mostly wasted in passively waiting for the primary node to fail.
```
Does it make sense to explain this, since it's not what Mule uses?

Cristian-Venticinque replied:
Please check if 981e4f0 is ok.

aiannucci commented on modules/ROOT/pages/mule-high-availability-ha-clusters.adoc:
```diff
@@ -13,101 +16,118 @@ Contact your customer service representative about pricing for this feature.
+A high-reliability application must meet the following requirements:
 
 . Zero tolerance for message loss
 . A reliable underlying enterprise service bus (Mule)
```
The concept of Mule as an ESB is quite old, how can we rewrite this?

Cristian-Venticinque replied:
That line could be replace with the definition of Mule we have in the runtime overview:
* A reliable integration engine (Mule)

aiannucci commented on modules/ROOT/pages/mule-high-availability-ha-clusters.adoc:
```diff
@@ -13,101 +16,118 @@ Contact your customer service representative about pricing for this feature.
+** Because each clustered Mule runtime engine runs on a different network node, each instance receives only the socket-based traffic sent to its node. You need to apply clustering and load balancing to distribute incoming socket-based traffic among the clustered instances.
 ** Output to socket-based transports is written to a specific host/port combination. If the host/port combination is an external host, no special considerations apply. If it is a port on the local host, consider using that port on the load balancer instead to better distribute traffic among the cluster.
 * Listener-based
 ** Listener-based transports fully support multiple readers and writers. No special considerations apply either to input or to output.
```
Review everything that talks about transports

Cristian-Venticinque replied:
Done here: 6c24446

aiannucci commented on modules/ROOT/pages/mule-high-availability-ha-clusters.adoc:
```diff
@@ -215,17 +233,19 @@ Remember that application-level configuration overrides container-level configur
+In cases of high load with endpoints that do not support load balancing, applying the performance profile might degrade performance. If you are using a File-based transport with an asynchronous processing strategy, JMS topics, multicasting, or HTTP connectors without a load balancer, the high volume of messages entering a single node can cause bottlenecks, and thus it can be better for performanc ...
+====
 
-You can also choose to define a minimum number of machines required in a cluster for it to remain in an operational state. This grants you a consistency improvement. See  xref:creating-and-managing-a-cluster-manually.adoc#quorum-management[Quorum Management].
```
Why are we removing the link to the Quorum management documentation?

Cristian-Venticinque replied:
It was moved to the See Also section, we have to avoid inline links due to style requirements

Commit:
Rephrased Mule behavior as a node.

Commit:
Clarified requirements when using multicast.

Commit:
Rewrote details and benefits of the active-active model.

Commit:
Removed "physical" as required during review.

Commit:
Replaced instances of "transport" with "connector".

dhirsch1 commented on modules/ROOT/pages/mule-high-availability-ha-clusters.adoc:
```diff
@@ -69,7 +69,9 @@ If multiple instances of the same application are deployed, they will fail becau
 * Mule high availability (HA) requires a reliable network connection between servers to maintain synchronization between the nodes in the cluster.
 * You must keep the following ports open to set up a Mule cluster: port `5701` and port `54327`.
-* You need to enable the multicast IP address: `224.2.2.3`. Multicast is used to discover new cluster members.
+* In case you configure your cluster to use multicast, you need to enable the multicast IP address: `224.2.2.3`.
```
"If you configure..." I would also put the last sentence back as a second sentence in the bullet.

dhirsch1 commented on modules/ROOT/pages/mule-high-availability-ha-clusters.adoc:
```diff
@@ -4,7 +4,10 @@ include::_attributes.adoc[]
+[IMPORTANT]
+Clustering is not available in CloudHub, see xref:runtime-manager::cloudhub-fabric.adoc[CloudHub Fabric] for details on how workers can be shared or doubled to scale your application and provide high availability.
+
+A Cluster is a set of Mule runtime engines that acts as a unit. In other words, a cluster is a virtual server composed of multiple nodes. The nodes (Mule runtime engines) in a cluster communicate and share information through a distributed shared memory grid. This means that the data is replicated across memory in different machines.
```
I suggest putting (Mule runtime engines) in the first sentence, after the word "nodes." It breaks up the second sentence., Also, Cluster should be lowercase.

dhirsch1 commented on modules/ROOT/pages/mule-high-availability-ha-clusters.adoc:
```diff
@@ -13,103 +16,118 @@ Contact your customer service representative about pricing for this feature.
 
-* *JMS Topics*: All mule instances connect to the same JMS topic, which may lead to repeated processing of messages when scaling the non clustered Mule instance horizontally
+* JMS topics. +
+All Mule instances connect to the same JMS topic, which might lead to repeated processing of messages when scaling the non-clustered Mule instance horizontally.
```
"the repeated processing"

dhirsch1 commented on modules/ROOT/pages/mule-high-availability-ha-clusters.adoc:
```diff
@@ -13,103 +16,118 @@ Contact your customer service representative about pricing for this feature.
 
-* *JMS request-reply/request-response*: All Mule instances are listening for messages in the same response queue, this implies that a Mule instance might obtain a response that isn't correlated to the request it sent. This can result in incorrect responses or make a flow fail with timeout.
+* JMS request-reply/request-response. +
+All Mule instances are listening for messages in the same response queue, which implies that a Mule instance might obtain a response that isn't correlated to the request it sent and might result in incorrect responses or make a flow fail with timeout.
```
This is a long sentence. I would split it up. "This implies that..."

dhirsch1 commented on modules/ROOT/pages/mule-high-availability-ha-clusters.adoc:
```diff
@@ -13,103 +16,118 @@ Contact your customer service representative about pricing for this feature.
 
-* *Salesforce streaming API*: if multiple instances of the same application are deployed, they will fail since the API only supports a single consumer. No failover support in case the instance connected is stopped or crashes
+* Salesforce streaming API. +
+If multiple instances of the same application are deployed, they will fail because the API only supports a single consumer. There is no failover support in case the instance connected is stopped or crashes.
```
Change "in case" to "if"

dhirsch1 commented on modules/ROOT/pages/mule-high-availability-ha-clusters.adoc:
```diff
@@ -13,103 +16,118 @@ Contact your customer service representative about pricing for this feature.
-In an *active-passive* model, one node in a cluster acts as the *primary*, or active node, while the others are *secondary*, or passive nodes. The application in such a model runs on the primary node, and only ever runs on the secondary node if the first one fails. In this model, the processing power of the secondary node(s) is mostly wasted in passive waiting for the primary node to fail.
-
-In an *active-active* model, no one node in the cluster acts as the primary node; all nodes in the cluster support the application. This application in this model runs on all the nodes, even splitting apart message processing between nodes to expedite processing across nodes.
+Mule uses an active-active model to cluster Mule runtime engines. The benefit of this model over an active-passive approach is that your application runs in all nodes, splitting apart message processing between the other nodes in your cluster expediting processing.
```
"splitting message processing with the other nodes in your cluster, which expedites processing.

dhirsch1 commented on modules/ROOT/pages/mule-high-availability-ha-clusters.adoc:
```diff
@@ -13,103 +16,118 @@ Contact your customer service representative about pricing for this feature.
+== Queues
 
-You can set up a *VM queue* explicitly to *load balance* across Mule runtimes (nodes). Thus, if your entire application flow contains a sequence of child flows, the cluster can assign each successive child flow to whichever Mule runtime (node) happens to be available at the time. Potentially, the cluster can process a single message on multiple nodes as it passes through the VM endpoints in the a ...
+You can set up a VM queue explicitly to load balance across Mule runtime engines (nodes). Thus, if your entire application flow contains a sequence of child flows, the cluster can assign each successive child flow to whichever Mule runtime engine (node) happens to be available at the time. Potentially, the cluster can process a single message on multiple nodes as it passes through the VM endpoint ...
```
You don't have to repeat (node) in each sentence.

"The cluster can potentially..."

dhirsch1 commented on modules/ROOT/pages/mule-high-availability-ha-clusters.adoc:
```diff
@@ -13,103 +16,118 @@ Contact your customer service representative about pricing for this feature.
 . Highly reliable individual connections
 
-The feature known as *transactionality* tracks application event sequences to ensure that each message-processing step gets completed successfully, and therefore, no messages get lost or processed incorrectly. If a step fails, for some reason, the transactional mechanism rolls back all previous processing events, then restarts the message processing sequence again.
+The feature known as transactionality tracks application event sequences to ensure that each message-processing step completes successfully, and therefore, no messages get lost or processed incorrectly. If a step fails for some reason, the transactional mechanism rolls back all previous processing events, then restarts the message processing sequence again.
```
"message-processing" instead of "message processing" in the last sentence. Also, the "again" is redundant. I would remove it,

dhirsch1 commented on modules/ROOT/pages/mule-high-availability-ha-clusters.adoc:
```diff
@@ -13,103 +16,118 @@ Contact your customer service representative about pricing for this feature.
+Connectors such as JMS, VM, and JDBC provide built-in transactional support, thus ensuring that messages get processed reliably. For example, you can configure a transaction on an inbound JMS connection endpoint to remove messages from the JMS server only after the transaction has been committed. This practice ensures that the original message remains available for reprocessing if an error occurs ...
 
-You must use *XA transactions* to move messages between dissimilar transports that support transactions. This ensures that the Mule runtime commits associated transactions from all the dissimilar transports as a single unit. See xref:transaction-management.adoc[Transaction Management] for more information on XA and other types of transactions supported by Mule runtimes.
+You must use XA transactions to move messages between dissimilar Connectors that support transactions. This ensures that the Mule runtime engine commits associated transactions from all the dissimilar connectors as a single unit.
```
Capitalize connectors consistently - either initial capped or not. Check the doc to see how it is usually capitalized. I think it is probably lower case unless it is part of a connector name.

dhirsch1 commented on modules/ROOT/pages/mule-high-availability-ha-clusters.adoc:
```diff
@@ -215,17 +231,19 @@ Remember that application-level configuration overrides container-level configur
+====
 
-You can also choose to define a minimum number of machines required in a cluster for it to remain in an operational state. This grants you a consistency improvement. See  xref:creating-and-managing-a-cluster-manually.adoc#quorum-management[Quorum Management].
+You can also choose to define the minimum number of machines that a cluster requires to remain in an operational state. This configuration grants you a consistency improvement in performance.
```
"consistency improvement in performance doesn't make sense." Do you mean "...improves performance consistency?"

dhirsch1 commented on modules/ROOT/pages/mule-high-availability-ha-clusters.adoc:
```diff
@@ -215,17 +231,19 @@ Remember that application-level configuration overrides container-level configur
 //TODO: UPDATE
 TODO: CHK/FIX LINKS /mule-user-guide/v/3.9/evaluating-mule-high-availability-clusters-demo?
-To evaluate Mule's HA clustering capabilities first-hand, continue on to the xref:3.9@mule-runtime::evaluating-mule-high-availability-clusters-demo.adoc[Mule HA Demo]. Designed to help new users evaluate the capabilities of Mule High Availability Clusters, the Mule HA Demo Bundle teaches you how to use the Mule Management Console to create a cluster of Mule runtimes, then deploy an application to ...
+To evaluate Mule's HA clustering capabilities first-hand, continue on to the xref:3.9@mule-runtime::evaluating-mule-high-availability-clusters-demo.adoc[Mule HA Demo]. Designed to help new users evaluate the capabilities of Mule High Availability Clusters, the Mule HA Demo Bundle teaches you how to use the Mule Management Console to create a cluster of Mule runtime engines, then deploy an applica ...
```
I suggest changing "then" to "and then." Also, "Further" is not needed. Instead, you could leave the sentence as is, starting with "This demo..."

Commit:
Addressing editorial changes requested during review.

Commit:
Updated description of Mule as an integration engine.

Commit:
Added small clarification about using ARM to manage clusters.
